docs/accessing-data/optimizing-queries.mdx

```diff
@@ -0,0 +1,189 @@
+---
+title: "Optimizing Queries"
+sidebar_position: 30
+---
+
+Hubble has terabytes of data to explore&mdash;that’s a lot of data! With access to so much data at your fingertips, it is crucial to performance-tune your queries.
+
+One of the strengths of BigQuery is also its pitfall: you have access to tremendous compute capabilities, but you pay for what you use. If you fine-tune your queries, you will have access to powerful insights at the fraction of the cost of maintaining a data warehouse yourself. It is, however, easy to incur burdensome costs if you are not careful.
+
+## Best Practices
+
+### Pay attention to table structure.
+
+Large tables are partitioned and clustered according to common access patterns. Prune only the partitions you need, and filter or aggregate by clustered fields when possible.
+
+Joining tables on strings is expensive. Refrain from joining on string keys if you can utilize integer keys instead.
+
+Read the docs on [Viewing Metadata](/docs/accessing-data/viewing-metadata) to learn more about table metadata.
+
+#### Example - Profiling Operation Types
+
+Let’s say you wanted to profile the [types of operations](/docs/fundamentals-and-concepts/list-of-operations) submitted to the Stellar Network monthly.
+
+<CodeExample>
+
+```sql
+# Inefficient query
+select datetime_trunc(batch_run_date, month) as `month`,
+  type_string,
+  count(id) as count_operations
+from `crypto-stellar.crypto_stellar.history_operations`
+group by `month`,
+  type_string
+order by `month`
+```
+
+</CodeExample>
+
+The `history_operations` table is partitioned by `batch_run_date` and clustered by `transaction_id`, `source_account` and `type`. Query costs are greatly reduced by pruning unused partitions. In this case, you could filter out operations submitted before 2023. This reduces the query cost by 4x.
+
+<CodeExample>
+
+```sql
+# Prune out partitions you do not need
+select datetime_trunc(batch_run_date, month) as `month`,
+  type_string,
+  count(id) as count_operations
+from `crypto-stellar.crypto_stellar.history_operations`
+where batch_run_date >= '2023-01-01T00:00:00'
```

Comment:
something something ISOsomething date format and/or golang docs, for clarity on `batch_run_date` format?